px.py

```diff
@@ -407,6 +407,20 @@ def __init__(self, proxy_type, proxy_server_address):
                     principal = (urlparse.quote(State.username) + ":" +
                         urlparse.quote(pwd))
 
+            # Attempt to resolve CNAME record for the `proxy_server_address` in case the proxy server is being hosted behind a DNS load balancing mechanism
```

Comment:
Might not be the best place to do this since it will happen for every single transaction. Is it possible to do once?

Reply:
Good point. Once per "proxy refresh" should do I believe. I'll look into it